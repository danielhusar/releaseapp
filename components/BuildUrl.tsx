import React from 'react'
import { IEnvironment } from '~/@types/environment'

export default function BuildUrl({
  source_control,
  repository_full_name,
  commit_long,
  commit_short,
}: IEnvironment['current_build']) {
  return (
    <a
      href={`https://${source_control}/${repository_full_name}/commit/${commit_long}`}
      target="_blank"
      rel="noreferrer"
    >
      {commit_short}
    </a>
  )
}
