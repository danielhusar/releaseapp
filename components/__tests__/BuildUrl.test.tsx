import React from 'react'
import { render, environment } from './helpers'
import BuildUrl from '../BuildUrl'

describe('BuildUrl', () => {
  it('renders', () => {
    const { container } = render(<BuildUrl {...environment.current_build} />)
    expect(container).toMatchSnapshot()
  })
})
