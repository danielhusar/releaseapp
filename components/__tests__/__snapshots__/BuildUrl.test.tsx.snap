// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildUrl renders 1`] = `
<div>
  <a
    href="https://github.com/tmcclung/release-rails-starter/commit/7ca25300fd20b337d06b087a15ed4f856e23454d"
    rel="noreferrer"
    target="_blank"
  >
    7ca2530
  </a>
</div>
`;
